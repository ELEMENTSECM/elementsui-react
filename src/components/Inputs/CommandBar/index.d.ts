import * as React from 'react';

export interface CommandBarProps {
	/**
	 * HTML id tag of the root element
	 */
	htmlId?: string;
	/**
	 * Items to render
	 */
	items?: any[];
	/**
	 * I to render on the opposite side of those defined in items
	 */
	farItems?: any[];
	/**
	 * User-defined styling
	 */
	styles?: (...args: any[]) => any;
}

declare const CommandBar: React.SFC<CommandBarProps>;

export default CommandBar;
