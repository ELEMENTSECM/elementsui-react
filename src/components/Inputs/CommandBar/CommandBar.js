import * as React from 'react';
import PropTypes from 'prop-types';
import { CommandBar as FabricCommandBar } from 'office-ui-fabric-react/lib/CommandBar';
import { getStyles } from './CommandBar.styles';
import { classNamesFunction, customizable, styled } from 'office-ui-fabric-react/lib/Utilities';

/** CommandBar example */
export function CommandBar(props) {
	const { items, farItems, getStyles } = props;
	const classNames = classNamesFunction()(getStyles, props);
	return <FabricCommandBar className={classNames.root} items={items} farItems={farItems} />;
}

CommandBar.propTypes = {
	/** Items to render */
	items: PropTypes.array,
	/** I to render on the opposite side of those defined in items */
	farItems: PropTypes.array,
	/** User-defined styling */
	getStyles: PropTypes.func
};

export default styled(customizable('CommandBar', ['theme'])(CommandBar), getStyles);
