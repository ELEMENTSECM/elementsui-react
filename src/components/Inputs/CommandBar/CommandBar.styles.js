export const styles = props => {
	const { className, theme } = props;
	const { palette, semanticColors } = theme;

	return {
		root: [
			'ms-Button',
			{
				// place your styles here
			},
			className
		]
	};
};
