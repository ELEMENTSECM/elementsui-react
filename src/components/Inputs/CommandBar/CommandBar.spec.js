import React from 'react';
import renderer from 'react-test-renderer';
import CommandBar from './CommandBar';

describe('CommandBar', () => {
	test('displays `Foo` as text', () => {
		const leftSideCommandBarButtons = [
			{
				key: 'upload',
				name: 'Upload',
				icon: 'Upload',
				onClick: () => {
					alert('Clicked!');
				}
			}
		];
		const tree = renderer
			.create(<CommandBar id="testCommandBar" items={leftSideCommandBarButtons} />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
