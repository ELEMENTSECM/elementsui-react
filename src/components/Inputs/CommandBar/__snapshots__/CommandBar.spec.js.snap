// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommandBar displays \`Foo\` as text 1`] = `
<div
  className="ms-CommandBar root_6e89cb9f ms-Button root-37"
>
  <div
    aria-describedby={undefined}
    aria-labelledby={undefined}
    className="ms-FocusZone container_6e89cb9f"
    data-focuszone-id="FocusZone1"
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDownCapture={[Function]}
    role="menubar"
  >
    <div
      className="ms-CommandBar-primaryCommands primaryCommands_6e89cb9f"
    >
      <div
        className="ms-CommandBarItem item_6e89cb9f"
      >
        <button
          aria-disabled={undefined}
          aria-expanded={undefined}
          aria-haspopup={false}
          aria-label={undefined}
          aria-posinset={1}
          aria-setsize={1}
          className="ms-CommandBarItem-link itemLink_6e89cb9f"
          data-command-key="upload"
          icon="Upload"
          id="CommandBar0upload"
          name="Upload"
          onClick={[Function]}
          role="menuitem"
          title=""
        >
          <i
            aria-hidden={true}
            className="root-38 ms-CommandBarItem-icon itemIcon_6e89cb9f ms-CommandBarItem-iconColor itemIconColor_6e89cb9f"
            data-icon-name="Upload"
            role="presentation"
          />
          <span
            className="ms-CommandBarItem-commandText itemCommandText_6e89cb9f"
          >
            Upload
          </span>
        </button>
      </div>
    </div>
    <div
      className="ms-CommandBar-sideCommands sideCommands_6e89cb9f"
    />
  </div>
</div>
`;
